relayer/src/connection.rs

```diff
@@ -108,6 +96,90 @@ impl Connection {
         Ok(c)
     }
 
+    pub fn find(
+        a_client: ForeignClient,
+        b_client: ForeignClient,
+        conn_end_a: &IdentifiedConnectionEnd,
+    ) -> Result<Connection, ConnectionError> {
+        Self::validate_clients(&a_client, &b_client)?;
+
+        // Validate the connection end
+        if conn_end_a.end().client_id().ne(a_client.id()) {
+            return Err(ConnectionError::ConstructorFailed(format!(
+                "the client id in the connection end ({}) does not match the foreign client id ({})",
+                conn_end_a.end().client_id(), a_client.id()
+            )));
+        }
+        if conn_end_a
+            .end()
+            .counterparty()
+            .client_id()
+            .ne(b_client.id())
+        {
+            return Err(ConnectionError::ConstructorFailed(format!(
+                "the counterparty client id in the connection end ({}) does not match the foreign client id ({})",
+                conn_end_a.end().counterparty().client_id(), b_client.id()
+            )));
+        }
+        if !conn_end_a.end().state_matches(&State::Open) {
+            return Err(ConnectionError::ConstructorFailed(format!(
+                "the connection end is expected to be in state 'Open'; found state: {:?}",
+                conn_end_a.end().state()
+            )));
+        }
+        let b_conn_id = conn_end_a
+            .end()
+            .counterparty()
+            .connection_id()
+            .cloned()
+            .ok_or_else(|| {
+                ConnectionError::ConstructorFailed(format!(
+                    "the connection end has no connection id field in the counterparty: {:?}",
+                    conn_end_a.end().counterparty()
+                ))
+            })?;
+
+        let c = Connection {
+            delay_period: 0, // TODO: Unclear if we should add mechanism to check the delay period
```

Comment:
Reminder for this TODO

Reply:
Will implement this with https://github.com/informalsystems/ibc-rs/issues/640. Let's leave the TODO (unless it's problematic to keep it around.)